### ISSUE TITLE
"swapchain image" vs "presentable image"

### ISSUE THREAD

**Issue by krOoze** [KhronosGroup/Vulkan-Docs#2005]
The specification has two terms: "swapchain image" and "presentable image". I am not sure what the distinction is. Might be the same thing, and the nomenclature might need to be harmonized. "swapchain image" seems to be missing in glossary, and "presentable image" basically says images obtained from swapchain. So I would prefer only "swapchain image".

**Comment by tuket** [comment]
I think "swapchain image" is an image that belongs to a swapchain. That is, when you call `vkCreateSwapchainKHR` those images are created. Then you can get those images calling `vkGetSwapchainImagesKHR`.

The spec *uses* "presentable image" for the same thing. But I think it's a terrible term for it.

> Presentable image
> A VkImage object obtained from a VkSwapchainKHR used to present to a VkSurfaceKHR object.

The definition in the glossary makes it even more confusing. "obtain" is a synonym for "acquire" so one could think that "presentable images" refers to those that have been acquired though `vkAcquireNextImageKHR`. And actually the term "presentable" suggests exactly that IMHO, as you *can't* present an image that you have not acquired first. However, that's not how the term "presentable image" is used in the spec. For example:

> A swapchain is an abstraction for an array of presentable images that are associated with a surface.

**Comment by krOoze** [comment]
@tuket "Obtain object" is not synonym for acquire. Pretty sure that means getting an object from `vkCreate*` or `vkGet*`.

**Comment by tuket** [comment]
I mean the verb "obtain" and "acquire" are synonyms in English: https://www.wordreference.com/synonyms/acquire

Or at least they have very similar meanings. English is not my native language though.

**Comment by krOoze** [comment]
@tuket In a specification it matters how terms are introduced, and not so much what is a synonym of what. "Acquire" is a very specific keyword there. Albeit, it is overloaded with something like four different things, because we already run out of other english synonyms to use.

How "acquire" is introduced is "acquire for use". So it is more of a shortcut for that or for the specific command.

**Comment by tuket** [comment]
Yes, I agree that "acquired" is used with a very specific meaning though the spec.

It's "obtain" that confused me a bit.

But reading more the spec, I realize that it's *mostly* (but not always) used as a synonym for "get" functions.

Examples:

```
To obtain the array of presentable images associated with a swapchain, call:
VkResult vkGetSwapchainImagesKHR ...

... vkGetDeviceProcAddr should be used to obtain function pointers for commands ...

Function pointers for all Vulkan commands can be obtained with the command:
PFN_vkVoidFunction vkGetInstanceProcAddr( ....
```

I guess as you read the spec you start getting a sense of how words are used.

Using the word "gotten" would be (IMHO) more consistent term for objects that are obtained get a "get" function. But this is probably being too picky :)

Anyways, back to the original question. Yes, I think "presentable image" is the same as "swapchain image". And I also prefer the term "swapchain image" for images that belong to a swapchain.

**Comment by stonesthrow** [comment]
Terminology is not always worked out from the beginning and much of it evolves as we delve into what do we mean exactly, and the details of corner cases.
We appreciate help in identifying these issues so that we can be more precise.
